### ISSUE TITLE
Closing last tab doesn't close the broswer window

### ISSUE THREAD

**Issue by umeshn1** [zen-browser/desktop#2537]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

Closing last tab doesn't close the browser window.

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.17

### What platform are you seeing the problem on?

Linux

**Comment by stripedgoat** [comment]
This is intended behavior.

**Comment by ur-fault** [comment]
Is it possible to disable it?

**Comment by Contade** [comment]
I like that behavior.

**Comment by ander-710** [comment]
> Is it possible to disable it?

You can go to `about:config` and set the value of `browser.tabs.closeWindowWithLastTab` to true

**Comment by bthornton191** [comment]
`browser.tabs.closeWindowWithLastTab` has no effect. Closing the last tab doesn't close the browser window whether `browser.tabs.closeWindowWithLastTab` is `true` or `false`.

**Comment by 314parley** [comment]
came in here looking for this issue, thank goodness I'm not crazy.

> This is intended behavior.

there is no way. if it is, there should probably be a setting for it. On vanilla Firefox, the default behavior is to close the window

**Comment by JackDyre** [comment]
> This is intended behavior.

my installation didn't start doing this before the latest update. if it is a feature, it must have gotten automatically turned on

**Comment by BiosPlus** [comment]
If you're running multi-window of any browser (FF, Chromium, Safari) and close the final tab of a window, it typically closes the window too; I was expecting the same behavior here.

**Comment by stripedgoat** [comment]
> If you're running multi-window of any browser (FF, Chromium, Safari) and close the final tab of a window, it typically closes the window too; I was expecting the same behavior here.

That's right, but case of Zen we have Workspaces. After closing your sites in one Workspace, you may very well want to switch to another Workspace. If the window would close when there's nothing on the particular Workspace, you'd have to click back on the app to bring it back into focus, and then switch to the next Workspace you wanted to.

The way it is now, basically the browser is waiting for a conscious decision of the user when he/she wants to close the browser window.

**Comment by ur-fault** [comment]
But still, wouldn't it be better to honor the user's settings, particularly in the about:config ?

I understand the point with workspaces, but as for me, I'm not using them yet, and would like behavior similar to other browsers.

**Comment by KonoTyran** [comment]
I'm also having this issue, and find it very unintuitive to not have the window close when I close the last tab. my `about:config` also has close window with last tab set to true.

**Comment by BiosPlus** [comment]
Maybe if the user only has one workspace it would be worth setting the behaviour to close the window? I can never really see myself using the workspace functionality.

**Comment by RayZ3R0** [comment]
If you do not intend to use workspaces, you can disable it in settings and this should not happen. Reopen if this still happens after disabling workspaces.

**Comment by gaardhus** [comment]
This still happens when I have a pinned tab, even though I have disabled workspaces.

**Comment by clembu** [comment]
Now that workspaces are default and core to the browser, this should definitely be addressed.

**Comment by Gambloide** [comment]
**# Potentially Off-Topic**
The current behavior (v1.6b) doesn't align with the arguments provided in this thread as I understand them, even for workspace users.

**## Why?**
1. do not have any workspaces configured 
2. open two browser windows with one tab each (both will occupy the 'default workspace' now)
3. hit `[Ctrl + w]` or your configured equivalent in either window

**### Actual result:**
Zen replaces your tab with a new tab; both windows stay open.

**### Expected result:**
The interacted with window closes because there is still another window of the same workspace open; one window remains.

So this seems somewhat broken even for users actually utilizing workspaces. Though it might be considered a separate issue since it is specifically about multiple windows of the same workspace and not about any window in general.

---

**# Completely on topic**
v1.6b goes against my muscle-memory for how to interact with ANY application. Zen is the only application that I am aware of reverting user instructions to close a tab instead of adhering to it. 

IMO, the current behavior makes no sense if a user has only a single workspace (usually the default workspace) configured either way and should be reconsidered.

If there is some use case I am not aware of that would make that behavior still desirable even with only a single workspace configured, I feel `browser.tabs.closeWindowWithLastTab` in `about:config` should override the current behavior going forward, now that workspaces cannot be disabled anymore.

**Comment by mr-cheffy** [comment]
Please do vote here https://www.reddit.com/r/zen_browser/comments/1hyuao1/discussion_about_how_we_handle_workspaces/

**Comment by CypElf** [comment]
Even if you decide to keep the current behavior by default, that is of opening a new tab again in the current workspace when the last tab is closed no matter if there's another workspace or not, I think having the option to override it with `browser.tabs.closeWindowWithLastTab` for people not happy with this choice wouldn't be an issue. Currently, being forced to adhere to this way of managing the last tab without the possibility to do anything about it, is, I believe, annoying for a lot of users.
